app/controllers/actions/AuthAction.scala

```diff
@@ -43,41 +43,32 @@ class AuthActionImpl @Inject()(
   implicit override val executionContext: ExecutionContext = mcc.executionContext
   override val parser: BodyParser[AnyContent] = mcc.parsers.defaultBodyParser
 
-  lazy val loginUrl = config.getOptional[String]("urls.login").getOrElse(throw new Exception("Missing login url configuration"))
-  lazy val continueLoginUrl =
+  private lazy val loginUrl = config.getOptional[String]("urls.login").getOrElse(throw new Exception("Missing login url configuration"))
+  private lazy val continueLoginUrl =
     config.getOptional[String]("urls.loginContinue").getOrElse(throw new Exception("Missing continue login url configuration"))
 
   override protected def refine[A](request: Request[A]): Future[Either[Result, AuthenticatedRequest[A]]] = {
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
     authorised(SignedInUser.authorisationPredicate)
-      .retrieve(
-        credentials and
-          name and
-          email and
-          affinityGroup and
-          internalId and
-          allEnrolments
-      ) {
-
-        case Some(credentials) ~ Some(name) ~ email ~ affinityGroup ~ Some(identifier) ~ enrolments if isEoriAllowListed(enrolments) =>
-          val signedInUser = SignedInUser(credentials, name, email, affinityGroup, identifier, enrolments)
-          Future.successful(Right(AuthenticatedRequest(request, signedInUser)))
-
-        case _ =>
-          //TODO Change this msg
-          throw new UnauthorizedException("Unable to retrieve internal Id")
-
-      } recover {
-      case _: NoActiveSession => Left(Redirect(loginUrl, Map("continue" -> Seq(continueLoginUrl))))
-      case _                  => Left(Redirect(routes.UnauthorisedController.onPageLoad()))
-    }
-  }
-
-  private def isEoriAllowListed(enrolments: Enrolments): Boolean = {
-    val eoriOpt = enrolments.getEnrolment("HMRC-CUS-ORG").flatMap(_.getIdentifier("EORINumber"))
-
-    eoriOpt.exists(eori => eoriAllowList.allows(eori.value))
+      .retrieve(allEnrolments) { allEnrolments: Enrolments =>
+        val eoriOpt: Option[EnrolmentIdentifier] = allEnrolments.getEnrolment(cdsEnrolmentName).flatMap(_.getIdentifier(eoriIdentifierKey))
+
+        if (eoriOpt.isDefined) {
+
+          if (eoriOpt.exists(eori => eoriAllowList.allows(eori.value))) {
+            val signedInUser = SignedInUser(eoriOpt.get.value, allEnrolments)
+            Future.successful(Right(AuthenticatedRequest(request, signedInUser)))
+          } else {
+            throw new UnauthorizedException("User is not authorized to use this service")
+          }
+        } else
+          throw InsufficientEnrolments()
```

Comment:
NIT: Personal taste probably, but I think the match style (Some(eori) if eoriAllowList =>, Some(_) =>, None =>) is more readable than nested if else statements. :slightly_smiling_face:

Reply:
To be fair I don't like this implementation too much :P Thank you for suggestion, I'll have a look.